apps/homescreen/test/unit/grid_test.js

```diff
@@ -40,9 +40,16 @@ var mocksHelperForGrid = new MocksHelper([
   'getDefaultIcon',
   'Bookmark',
   'BookmarksManager',
-  'BookmarksDatabase'
+  'BookmarksDatabase',
+  'L10n'
 ]).init();
 
+// Pre-mock mozL10n as request.js inits using mozL10n...
+var realL10n = navigator.mozL10n;
+navigator.mozL10n = {
+  get: function() {}
+};
```

Comment:
I now see why it fails.  `GridManager` binds `_` [very early](https://github.com/mozilla-b2g/gaia/blob/a14de2c47999a65cff5c06d92625c3ef73e7fb8c/apps/homescreen/js/grid.js#L59), when `navigator.mozL10n` is still `undefined` in tests.  Pre-mocking helps a bit, but it relies on the fact that `requireApp('homescreen/js/grid.js');` is asynchronous and I'd prefer to avoid such racy code if possible.

Otherwise, `GridManager` is well-designed: all initialization happens explicitly in `GridManager.init` which is called in tests' `setupSuite`. Let's use that.

I suggest you remove the assignment to `_` in `GridManager` and instead use `navigator.mozL10n.get` everywhere in the js/grid.js file.  This will fix the `TypeError` because you won't be trying to access `navigator.mozL10n.get` when `navigator.mozL10n` is `undefined`.  OTOH, when `GridManager.init` is called, you will have already mocked `mozL10n` in `setupSuite` and any subsequent `navigator.mozL10n.get` calls will call into `MockL10n`.

Let me know what you think about this approach.